include/twin.h

```diff
@@ -619,7 +619,9 @@ twin_pixmap_t *twin_make_cursor(int *hx, int *hy);
  * dispatch.c
  */
 
-void twin_dispatch(void);
+struct _twin_context;
+
+void twin_dispatch(struct _twin_context *ctx);
```

Comment:
Why don't you use `twin_context_t`?

Reply:
The typedef of the struct `_twin_context` is defined after `twin_dispatch()` at the end of the header file. I'm not sure where should I move that struct to, so I just left it there and used its raw name.

Reply:
You can simply move the definition of `twin_context_t` in the earlier section of `include/twin.h`.